modules/ppcp-settings/resources/js/data/common/reducer.js

```diff
@@ -39,7 +43,10 @@ const commonReducer = createReducer( defaultTransient, defaultPersistent, {
 		setPersistent( state, action ),
 
 	[ ACTION_TYPES.HYDRATE ]: ( state, payload ) =>
-		setPersistent( state, payload.data ),
+		setPersistent( state, {
+			...payload.data,
+			flags: payload.flags,
+		} ),
```

Comment:
This logic is not entirely correct.

The "flags" (or "wooSettings") should not be part of the Persistent object: The "setPersistent" method should only store data that is input by the user (form fields) and that should be saved/changed on the server.

The country/currency are read-only flags and must be manually added tot the new state object, like this:

```js
// ...
	[ ACTION_TYPES.HYDRATE ]: ( state, payload ) => {
		const newState = setPersistent( state, payload.data ); // this sets "persistant" items to the state.data object

		// Here we set read-only data to the state.wooSettings object. This object should be only accessible via a new selector
		if ( payload.wooSettings ) {
			newState.wooSettings = { ...newState.wooSettings, ...payload.wooSettings };
		}

		return newState;
	},
// ...
```

Reply:
Hello @stracker-phil, I've applied all your suggested changes in the latest commit. Thanks 👍🏻